packages/menu/src/MenuItem/MenuItem.spec.tsx

```diff
@@ -0,0 +1,145 @@
+
+      expect(menuItem.tagName.toLowerCase()).toBe('a');
+    });
+
+    test('div when "as" is supplied', () => {
+      const { getByTestId } = render(
+        <MenuItem data-testid="menu-item" as="div">
+          Test Content
+        </MenuItem>,
+      );
+      const menuItem = getByTestId('menu-item');
+
+      expect(menuItem.tagName.toLowerCase()).toBe('div');
+    });
+
+    test('fires onClick callback when clicked', () => {
+      const clickHandler = jest.fn();
+
+      const { getByTestId } = render(
+        <MenuItem data-testid="menu-item" onClick={clickHandler} />,
+      );
+      const menuItem = getByTestId('menu-item');
+      userEvent.click(menuItem);
+      expect(clickHandler).toHaveBeenCalledTimes(1);
+    });
+
+    test(`renders className in the MenuItem container's classList`, () => {
+      const { getByTestId } = render(
+        <MenuItem data-testid="menu-item" className="menu-item-class-name" />,
+      );
+      const menuItem = getByTestId('menu-item');
+      expect(menuItem.classList.contains('menu-item-class-name')).toBe(true);
+    });
+
+    test('renders MenuItem inside button tag by default', () => {
+      const { getByTestId } = render(<MenuItem data-testid="menu-item" />);
+      const menuItem = getByTestId('menu-item');
+      expect(menuItem.tagName.toLowerCase()).toBe('button');
+    });
+
+    test('renders inside of an `a` instead of a `button` tag, when `href` prop is supplied', () => {
+      const { getByTestId } = render(
+        <MenuItem href="https://mongodb.design" data-testid="menu-item" />,
+      );
+      const menuItem = getByTestId('menu-item');
+      expect(menuItem.tagName.toLowerCase()).toBe('a');
+    });
+
+    test('renders with correct target and rel values when set', () => {
+      const { getByTestId } = render(
+        <MenuItem
+          as="a"
+          href="https://mongodb.design"
+          target="_blank"
+          rel="help"
+          data-testid="menu-item"
+        />,
+      );
+      const menuItem = getByTestId('menu-item');
+
+      expect((menuItem as HTMLAnchorElement).target).toBe('_blank');
+      expect((menuItem as HTMLAnchorElement).rel).toBe('help');
+    });
+
+    test('renders as `div` tag when the "as" prop is set', () => {
+      const { getByTestId } = render(
+        <MenuItem data-testid="menu-item" as="div" />,
+      );
+      const menuItem = getByTestId('menu-item');
+      expect(menuItem.tagName.toLowerCase()).toBe('div');
+    });
+
+    test('has the `aria-current` attribute when active', () => {
+      const { getByTestId } = render(
+        <MenuItem data-testid="menu-item" active={true} />,
+      );
+      const menuItem = getByTestId('menu-item');
+      expect(menuItem).toHaveAttribute('aria-current', 'true');
+    });
+  });
+
+  /* eslint-disable jest/no-disabled-tests */
+  describe('types behave as expected', () => {
+    test.skip('Accepts string as `as` prop', () => {
```

Comment:
should we be checking that this renders the proper HTML element?

Reply:
We're checking that on L27. This is just the TS test